ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -417,6 +377,293 @@ Map<String, DataSource> getDataSources() {
+          if (expirationTime != null && expirationTime.before(deletionTime)) {
+            deletionTime = expirationTime;
+          }
+
+          if (compareTimestamp(deletionTime, currentTime) <= 0) {
+            logger.trace("LIST: Blob is not available, ignoring in list response; account='{}', container='{}', name='{}'",
+                accountName, containerName, blobName);
+          } else {
+            entries.add(
+                new NamedBlobRecord(accountName, containerName, blobName, blobId, timestampToMs(deletionTime)));
+          }
+        }
+        return new Page<>(entries, nextContinuationToken);
+      }
+    }
+  }
+
+  private Page<NamedBlobRecord> run_list_v2(String accountName, String containerName, String blobNamePrefix,
+      String pageToken, short accountId, short containerId, Connection connection) throws Exception {
+    try (PreparedStatement statement = connection.prepareStatement(LIST_QUERY_V2)) {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, blobNamePrefix + "%");
+      statement.setString(4, pageToken != null ? pageToken : blobNamePrefix);
+      statement.setInt(5, config.listMaxResults + 1);
+      try (ResultSet resultSet = statement.executeQuery()) {
+        String nextContinuationToken = null;
+        List<NamedBlobRecord> entries = new ArrayList<>();
+        long currentTime = System.currentTimeMillis();
+        int resultIndex = 0;
+        while (resultSet.next()) {
+          if (resultIndex++ == config.listMaxResults) {
+            nextContinuationToken = resultSet.getString(1);
+            break;
+          }
+          String blobName = resultSet.getString(1);
+          String blobId = Base64.encodeBase64URLSafeString(resultSet.getBytes(2));
+          long version = resultSet.getLong(3);
+          Timestamp deletionTime = resultSet.getTimestamp(4);
+
+          if (compareTimestamp(deletionTime, currentTime) <= 0) {
+            logger.trace("LIST: Blob is not available, ignoring in list response; account='{}', container='{}', name='{}'",
+                accountName, containerName, blobName);
+          } else {
+            entries.add(
+                new NamedBlobRecord(accountName, containerName, blobName, blobId, timestampToMs(deletionTime), version));
+          }
+        }
+        return new Page<>(entries, nextContinuationToken);
+      }
+    }
+  }
+
+  private PutResult run_put(NamedBlobRecord record, short accountId, short containerId, Connection connection)
+      throws Exception {
+    // 1. Pulling existing row content so that we can get the min(expires_ts, deleted_ts) of existing and new records
+    NamedBlobRecord existingRow;
+    try {
+      existingRow = run_get(record.getAccountName(), record.getContainerName(), record.getBlobName(),
+          GetOption.Include_All, accountId, containerId, connection);
+    } catch (Exception e) {
+      existingRow = null;
+    }
```

Comment:
As per the comment above that code:
`// 1. Pulling existing row content so that we can get the min(expires_ts, deleted_ts) of existing and new records`
Suppose there is existing row with:
x, y, z, ... deleted_ts_1, ...

And new data to be inserted has min (expires_ts/deleted_ts) as 'deleted_ts_2'.
Then we need to insert the data as min (deleted_ts_1, deleted_ts_2), so we need that get query.